internal/manager/config.go

```diff
@@ -202,7 +205,10 @@ func (c *Config) GetKongClient(ctx context.Context) (*kong.Client, error) {
 		return nil, err
 	}
 
-	return adminapi.GetKongClientForWorkspace(ctx, c.KongAdminURL, c.KongWorkspace, httpclient)
+	clientSetup.Lock()
+	client, err := adminapi.GetKongClientForWorkspace(ctx, c.KongAdminURL, c.KongWorkspace, httpclient)
+	clientSetup.Unlock()
```

Comment:
I'm sorry I really should have thought about this more and put more emphasis on my point from before about putting this as close to the source as possible. Moving the lock here leaves a similar problem as before: we have a public function which the caller has to reason about its thread-safety type issues. We should move this lock down one level deeper: the `GetKongClientForWorkspace()` function should lock right before `internal/adminapi/client.go:91` where the `exists` check is done, and then direct callers of that function (including manager code) will no longer have the race condition.

Reply:
b034eaa1